pkg/cloud/api/path.go

```diff
@@ -71,7 +91,36 @@ func (p Path) Equal(other Path) bool {
 	return true
 }
 
-// HasPrefix returns true if prefix is the prefix of this path.
+// Match the given path, interpreting wildcard matching for the comparison. Use
+// Equal() instead to compare Paths for absolute equality (no wildcard
+// interpretation).
+func (p Path) Match(other Path) bool {
+	if len(p) != len(other) {
+		return false
+	}
+	for i := range p {
+		// Wildcard match for slices and maps indices
+		if p[i] == anySliceIndex && isSliceIndex(other[i]) {
+			continue
+		}
+		if other[i] == anySliceIndex && isSliceIndex(p[i]) {
+			continue
+		}
+		if p[i] == anyMapIndex && isMapIndex(other[i]) {
+			continue
+		}
+		if other[i] == anyMapIndex && isMapIndex(p[i]) {
+			continue
+		}
+		if p[i] != other[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// HasPrefix returns true if prefix is the prefix of this path. Prefix uses
+// Match() semantics for wildcards.
```

Comment:
Let make this predicate symmetric and add (`... wildcards. This predicate is symmetric.`) unless we have a good reason to make it non-symmetric, e.g. there is a use case. This is easier to use and understand and is implied by the statement, this uses Match() semantics, as Match is symmetric.

Reply:
I don't really think that symmetric is a good word here. 
For me symmetric means that p.HasPrefix(prefix) == prefix.HasPrefix(p) but this condition will not be true because 
in `HasPrefix` we expect that path length >= prefix length and that means that this function is not symmetric. 
By symmetric you mean that both prefix and that can have wildcards?
IMO prefix should be generic (and can use wildcard because we don't care about real indexes) and path should be a regular path extracted from the object so it has to have real indexes and map's key. 

We cannot use path.Match here because Match check path lengths.
I extracted isMatch function which compares the elements (and should be used in for loop) in case we wanted to use it here.

Reply:
make sense -- this is not symmetric